### PR TITLE
Bug 1216546: Progressive loading of entities

### PR DESCRIPTION
**Pull request by mathjazz**
This brings the effort stated in #335 as mostly backend code to frontend and adds some fixes.

**TODO**
- [x] Fix stats when All resources or subpages with multiple resources selected.
- [x] Fix homepage: flickering on load.
- [x] Fix homepage: unable to switch to other project.
- [x] Fix history (back/forward).

### CODE REVIEW

**Comment by mathjazz** [comment]
Phew, that took me way longer than I thought! Lots of things were broken and many things should now work better than before. Here's a short summary of tasks:

* Prettier entity loading indicator.
* Removed Search tab from the Parts menu and locale-project page.
* Added All Resources menu entry instead.
* Fixed history: navigating back/forward between selected locale/project/parts.
* Fixed flickering on inplace page load, e.g. homepage.
* Fixed switching between in-place and non-in-place projects.
* Fixed has suggestions filter.
* Removed client-side filters and search.
* Do not trigger search with modifier keys etc. - only if value changes (old problem, but more visible with backend search).
* Don't show out of date editor content on search and filter (select frist entity or hide editor if no match).
* Focus & unfocus iframe properly so history navigation (back/forward) doesn't trigger in the iframe.

TODO:
- [ ] https://pastebin.mozilla.org/8864713: Search inside missing returns translated & search inside translated doesnt return anything (homepage - translated - github).

**Comment by mathjazz** [comment]
@jotes r?

File: pontoon/base/models.py [hunk 1]
```diff
@@ -1313,11 +1313,11 @@ def aggregate_stats(self, instance):
 
         instance.save()
 
-    def stats(self, paths, locale):
+    def stats(self, project, paths, locale):
         """
         Returns statistics for the given paths and locale.
         """
-        return self.get(locale=locale, resource__path__in=paths).stats
+        return self.get(resource__project=project, resource__path__in=paths, locale=locale).stats
```
Comment (jotes) [inline pontoon/base/models.py]: good catch.

**Comment by jotes** [comment]
:+1: :100: :dancers: :fireworks:

**Comment by jotes** [comment]
r+

**Pull request merged**